### DISCUSSION TITLE
Computing distance-r closed neighborhoods

### DISCUSSION THREAD

**Discussion by mogproject** [networkit/networkit#793] [Q&A] [answered]
Does NetworKit provide an efficient way to compute the distance-r closed neighborhood of all vertices in a given (undirected) graph? We want to obtain a set of vertices whose distance from vertex v is less than or equal to r, for all v in the graph.

We may assume that the given graph is sparse, and r is very small (like less than 10). We want some parallelism during computation. Equivalently, how can we construct a distance-r (transitive) closure of a given graph?

Looking at code, I guess `NeighborhoodFunction` in `distance.pyx` might be useful for our purpose, but it seems to compute distances between all vertex pairs, including those with a distance greater than r.

Any suggestions would be appreciated.

**Comment by angriman** [comment, marked as answer]
As far as I know, NetworKit does not provide an implementation for such a feature. On the other hand, a trivial implementation of this algorithm this does not look too hard to write, one could simply run a pruned BFS/Dijkstra from each vertex in parallel.

**Reply by mogproject** [reply]
Yes, I think simple BFS works for this purpose. I'm trying several implementations like this: https://gist.github.com/mogproject/3dab38352f4c64dbe6f31bd697f34afd
Thank you!
